net/rhp/objects.go

```diff
@@ -586,3 +580,430 @@ func (r *RPCWriteResponse) DecodeFrom(d *types.Decoder) {
 }
+
+// RPCSettingsResponse contains the JSON-encoded settings for a host.
+type RPCSettingsResponse struct {
+	Settings []byte
+}
+
+// MaxLen returns the maximum encoded length of an object. Implements rpc.Object.
+func (r *RPCSettingsResponse) MaxLen() int {
+	// JSON encoded settings should result in < 2 KiB
+	return 3000
+}
+
+// EncodeTo encodes a RPCSettingsResponse to an encoder. Implements types.EncoderTo.
+func (r *RPCSettingsResponse) EncodeTo(e *types.Encoder) {
+	e.WriteBytes(r.Settings)
+}
+
+// DecodeFrom decodes a RPCSettingsResponse from a decoder. Implements types.DecoderFrom.
+func (r *RPCSettingsResponse) DecodeFrom(d *types.Decoder) {
+	r.Settings = d.ReadBytes()
+}
+
+// RPCTrackingResponse is an empty RPC response to signal success.
+type RPCTrackingResponse struct{}
+
+// MaxLen returns the maximum encoded length of an object. Implements rpc.Object.
+func (r *RPCTrackingResponse) MaxLen() int {
+	return 0
+}
+
+// EncodeTo encodes a RPCTrackingResponse to an encoder. Implements types.EncoderTo.
+func (r *RPCTrackingResponse) EncodeTo(e *types.Encoder) {
+}
+
+// DecodeFrom decodes a RPCTrackingResponse from a decoder. Implements types.DecoderFrom.
+func (r *RPCTrackingResponse) DecodeFrom(d *types.Decoder) {
+}
+
+// RPCExecuteProgramRequest is the request for the RPC method "execute".
+type RPCExecuteProgramRequest struct {
+	// FileContractID is the id of the filecontract we would like to modify.
+	FileContractID types.ElementID
+	// Instructions are the instructions to be executed.
+	Instructions []Instruction
+	// ProgramDataLength is the length of the programData following this
+	// request.
+	ProgramDataLength uint64
+}
+
+// MaxLen returns the maximum encoded length of an object. Implements rpc.Object.
+func (req *RPCExecuteProgramRequest) MaxLen() int {
+	return defaultMaxLen
+}
+
+// EncodeTo encodes a RPCExecuteProgramRequest to an encoder. Implements types.EncoderTo.
+func (req *RPCExecuteProgramRequest) EncodeTo(e *types.Encoder) {
+	req.FileContractID.EncodeTo(e)
+	e.WriteUint64(uint64(len(req.Instructions)))
+	for _, instr := range req.Instructions {
+		instr.EncodeTo(e)
+	}
+	e.WriteUint64(req.ProgramDataLength)
+}
+
+// DecodeFrom decodes a RPCExecuteProgramRequest from a decoder. Implements
+// types.DecoderFrom.
+func (req *RPCExecuteProgramRequest) DecodeFrom(d *types.Decoder) {
+	req.FileContractID.DecodeFrom(d)
+	req.Instructions = make([]Instruction, d.ReadUint64())
+	for i := range req.Instructions {
+		req.Instructions[i].DecodeFrom(d)
+	}
+	req.ProgramDataLength = d.ReadUint64()
+}
+
+// Payment specifiers are used to specify the payment type
+var (
+	PayByContract         = rpc.NewSpecifier("PayByContract")
+	PayByEphemeralAccount = rpc.NewSpecifier("PayByEphemAcc")
+)
+
+// WithdrawalMessage is the amount of money to deduct from the account to create
+// the RPC budget.
+type WithdrawalMessage struct {
+	AccountID types.PublicKey
+	Expiry    uint64
+	Amount    types.Currency
+	Nonce     [8]byte
```

Comment:
this is the sort of field that I'd want a comment on -- why does a withdrawal need a nonce?